### PR TITLE
Feat/target path profiles dir 

### PR DESCRIPTION
**Pull request by mdesmet**
## Checklist

- [ ] I have run this code and it appears to resolve the stated issue
- [ ] `README.md` updated and added information about my change

### CODE REVIEW

File: src/dbt_client/dbtCoreIntegration.ts [hunk 1]
```diff
@@ -527,6 +511,21 @@ export class DBTCoreProjectIntegration
...
+    const profilesDir = await this.python.lock(
+      (python) =>
+        python`default_profiles_dir(Path(${this.projectRoot.fsPath}))`,
+    );
+    return profilesDir;
+  }
+
+  private async getTargetDirectory(): Promise<string> {
+    const targetDir = await this.python.lock(
+      (python) => python`target_path(${this.projectRoot.fsPath})`,
```
Comment (mdesmet) [inline src/dbt_client/dbtCoreIntegration.ts:524]: This line before was `python`target_path(Path(${this.projectRoot.fsPath}))`,`, That's the only place where we were using a pathlib PosixPath. It was not necessary and for some reason it didn't throw any errors in local testing.

File: dbt_integration.py [hunk 2]
```diff
@@ -169,25 +167,48 @@ def wrapper(
...
-def find_package_paths(project_directories, profiles_dir_override):
+def default_profiles_dir(project_dir):
+    if "DBT_PROFILES_DIR" in os.environ:
+        profiles_dir = os.path.expanduser(os.environ["DBT_PROFILES_DIR"])
+        if os.path.isabs(profiles_dir):
+            return os.path.normpath(profiles_dir)
+        return os.path.join(project_dir, profiles_dir)
+    project_profiles_file = os.path.normpath(os.path.join(project_dir, "profiles.yml"))
+    return (
+        project_dir
```
Comment (suryaiyer95) [inline dbt_integration.py:179]: We need to convert this to string right?

File: src/dbt_client/dbtCoreIntegration.ts [hunk 3]
```diff
@@ -527,6 +511,21 @@ export class DBTCoreProjectIntegration
     return macroPaths;
   }
 
+  private async findProfilesDirectory(): Promise<string> {
+    const profilesDir = await this.python.lock(
+      (python) =>
+        python`default_profiles_dir(Path(${this.projectRoot.fsPath}))`,
```
Comment (suryaiyer95) [inline src/dbt_client/dbtCoreIntegration.ts:517]: I think, this is the one causing the issue.

File: dbt_integration.py (hunk 2 above)
Comment (mdesmet) [reply dbt_integration.py:179]: Nope this should all be strings. os.path operations operatate on strings.

**Review by suryaiyer95 — COMMENTED** [review]
Tested cases and passed

1. Target path override
- dbt_project.yml 
     - target-path: 'compiled_files'
     - target-path: '../compiled_files'
-  DBT_TARGET_PATH environment variable


2. Profiles dir override
- DBT_PROFILES_DIR
      - '../../.db/profiles.yml'
      - './profiles.yml'
      - '~/.dbt'

**Review by anandgupta42 — APPROVED** [review]
LGTM

**Pull request merged**